spec/liquid_tags/codepen_tag_spec.rb

```diff
@@ -17,10 +17,23 @@ def generate_new_liquid(link)
       Liquid::Template.parse("{% codepen #{link} %}")
     end
 
+    def page_with_codepen_tag
+      <<~HTML
+        <!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN" "http://www.w3.org/TR/REC-html40/loose.dtd">
+        <html xmlns="http://www.w3.org/1999/xhtml">
+          <body>
+            <iframe height="600" src="https://codepen.io/twhite96/embed/XKqrJX?height=600&amp;default-tab=result&amp;embed-version=2" scrolling="no" frameborder="no" allowtransparency="true" loading="lazy" style="width: 100%;">
+            </iframe>
+          </body>
+        </html>
+      HTML
+    end
```

Comment:
This is me attempting to move [`spec/support/fixtures/approvals/codepen_liquid_tag.approved.html`](https://github.com/forem/forem/blob/c0ea4e58787a1142ddb952fdbe2709f177158ec0/spec/support/fixtures/approvals/codepen_liquid_tag.approved.html) inline and out of the approvals fixture file.

Is this along the lines that you were thinking, @joshpuetz? (This actually doesn't work for some reason, I haven't investigated deeply yet but if you see what I'm doing wrong, please let me know 😕)

Reply:
Reasons this is failing:

1. The approvals gem seems to strip/ignore whitespace, this does not.
2. `page_with_codepen_tag` uses `&amp;` but `rendered_codepen_iframe ` includes an actual `&` character.

Here's what I did to get the spec to pass:

```diff
diff --git a/spec/liquid_tags/codepen_tag_spec.rb b/spec/liquid_tags/codepen_tag_spec.rb
index 93967a436..224a38cc3 100644
--- a/spec/liquid_tags/codepen_tag_spec.rb
+++ b/spec/liquid_tags/codepen_tag_spec.rb
@@ -18,11 +18,11 @@ def generate_new_liquid(link)
     end

     def page_with_codepen_tag
-      <<~HTML
+      <<~HTML.gsub(/\s+/, " ")
         <!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN" "http://www.w3.org/TR/REC-html40/loose.dtd">
         <html xmlns="http://www.w3.org/1999/xhtml">
           <body>
-            <iframe height="600" src="https://codepen.io/twhite96/embed/XKqrJX?height=600&amp;default-tab=result&amp;embed-version=2" scrolling="no" frameborder="no" allowtransparency="true" loading="lazy" style="width: 100%;">
+            <iframe height="600" src="https://codepen.io/twhite96/embed/XKqrJX?height=600&default-tab=result&embed-version=2" scrolling="no" frameborder="no" allowtransparency="true" loading="lazy" style="width: 100%;">
             </iframe>
           </body>
         </html>
@@ -32,8 +32,9 @@ def page_with_codepen_tag
     it "accepts codepen link" do
       liquid = generate_new_liquid(codepen_link)
       rendered_codepen_iframe = liquid.render
+      expected = rendered_codepen_iframe.gsub(/\s+/, " ")
       # Approvals.verify(rendered_codepen_iframe, name: "codepen_liquid_tag", format: :html)
-      expect(page_with_codepen_tag).to include(rendered_codepen_iframe)
+      expect(page_with_codepen_tag).to include(expected)
     end
```

But, I'm not sure if I see any real benefit to using this style of approval for the liquid tags at all. We're using `include` as a matcher, so nothing else in `page_with_codepen_tag` really matters. We might as well just check that `liquid.render` produces the expected output.

Reply:
Yeah, that's more or less what I was thinking @vaidehijoshi. And I agree with @citizen428: many of these refactors will lead to a point where it's easier to just do a build in rspec matcher!